compiler/luci/partition/include/luci/Partition.h

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (c) 2021 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef __LUCI_PARTITION_H__
+#define __LUCI_PARTITION_H__
+
+#include <luci/IR/Module.h>
+
+#include <memory>
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+namespace luci
+{
+
+/**
+ * @brief PartitionTable holds partition information
+ */
+struct PartitionTable
+{
+  std::vector<std::string> groups;
+  std::string def_group;
+
+  // assign by opcode: OPCODE=group
+  std::unordered_map<std::string /* OPCODE */, std::string /* group */> byopcodes;
+
+  // TODO add assign by OP name
+};
+
+/**
+ * @brief PartedModule holds partitioned module and group name
+ */
+struct PartedModule
+{
+  std::unique_ptr<Module> module;
+  // group name used to partition this module
+  std::string group;
+
+  // unique name(filename) of this module
+  std::string name;
```

Comment:
IMHO, `name`, `group` and `id` is really confusing me..
```cxx
struct PGraph
{
  std::vector<std::unique_ptr<PNode>> pnodes;
  std::string group;
  uint32_t id = 0;

  // I/O while partitioning
  std::vector<luci::CircleNode *> inputs;
  std::vector<luci::CircleNode *> outputs;
};
```
`PGraph` has a `id` attribute, which I thought was its identifier(unique), But `PartedModule` has a `name` attribute for its uniqueness. Are they(`id` and `name`) same meaning?

Reply:
I'm don't know why you mentioned `PGraph` here... as I wrote above, it has no direct relations...

Reply:
If you want to revise the file of `PGraph`, please post an issue or PR to reflect your thoughts.

Reply:
Well, I would if needed. Maybe things that are confusing me will be clear after more PRs are merged.